c/cmake_modules/AdbcDefines.cmake

```diff
@@ -92,6 +92,17 @@ elseif(CMAKE_CXX_COMPILER_ID STREQUAL "AppleClang"
       -Werror
       -Wno-unused-parameter)
   set(ADBC_C_CXX_FLAGS_PRODUCTION -Wall)
+
+  string(APPEND C_DEBUG_FLAGS " -Og -ggdb")
+  string(APPEND CXX_DEBUG_FLAGS " -Og -ggdb")
+  string(APPEND C_RELWITHDEBINFO_FLAGS " -Og -ggdb")
+  string(APPEND CXX_RELWITHDEBINFO_FLAGS " -Og -ggdb")
```

Comment:
How about using `-g3` instead of `-ggdb`?
`-g3` provides more debug information than `-ggdb`.
For example, we can use expand macro in gdb with `-g3`.